src/index.js

```diff
@@ -92,49 +92,52 @@ IdTokenVerifier.prototype.verify = function(token, nonce, cb) {
   var nbf = jwt.payload.nbf;
   var tnonce = jwt.payload.nonce || null;
   /* eslint-enable vars-on-top */
+  var _this = this;
 
-  if (this.issuer !== iss) {
-    return cb(
-      new error.TokenValidationError('Issuer ' + iss + ' is not valid.'),
-      false
-    );
-  }
-
-  if (this.audience !== aud) {
-    return cb(
-      new error.TokenValidationError('Audience ' + aud + ' is not valid.'),
-      false
-    );
-  }
+  this.getRsaVerifier(iss, kid, function(err, rsaVerifier) {
```

Comment:
the issuer (`iss` var) should not be obtained from the payload before verifying the signature, but rather given explicitly by the user